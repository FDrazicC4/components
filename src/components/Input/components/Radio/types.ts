import React from 'react';
import { StyledComponent } from '@components/types';


export type LabelPlacements = (
    | 'top'
    | 'bottom'
    | 'start'
    | 'end'
);


export interface Props extends StyledComponent {

    /**
     * If `true`, the radio is checked.
     */
    isChecked?: boolean,

    /**
     * If `true`, the radio will be disabled.
     */
    isDisabled?: boolean,

    /**
     * If `true`, the radio will be required.
     */
    isRequired?: boolean,

    /**
     * Name attribute of the input element.
     */
    name?: string,

    /**
     * The value of the radio. The DOM API casts this to a string.
     */
    value?: any,

    /**
     * The label of the input
     */
    label?: React.ReactNode,

    /**
     * The placement of the label
     * @default 'end'
     */
    labelPlacement?: LabelPlacements,

}
