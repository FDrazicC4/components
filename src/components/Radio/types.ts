import { RadioProps } from '@material-ui/core/Radio';


export interface Props extends RadioProps {}
