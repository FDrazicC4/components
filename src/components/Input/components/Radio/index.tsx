import React, { ReactElement } from 'react';
import clsx from 'clsx';
import InputBase from '../InputBase';
import { Props } from './types';

function RadioCustomControl(): ReactElement {
    return (
        <div className="brew-Input__customControl" />
    );
}

function Radio({
    className,
    disabled = false,
    fullWidth = false,
    inputLabelClassName,
    ...otherProps
}: Props): ReactElement {

    const classes = clsx(
        'brew-Input--radio',
        className,
    );

    return (
        <InputBase
            className={classes}
            customControl={<RadioCustomControl />}
            disabled={disabled}
            fullWidth={fullWidth}
            inputLabelClassName={inputLabelClassName}
            type="radio"
            {...otherProps}
        />
    );

}


export default Radio;
