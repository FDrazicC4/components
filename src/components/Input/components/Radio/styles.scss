@import '../../../../styles/utilities';

.wrapper {
    display: inline-flex;
    overflow: hidden;

    input {
        position: absolute;
        left: -9999px;
    }
}

.radio {
    height: size(xx-large);
    width: size(xx-large);
    display: inline-flex;
    border-radius: 100%;
    border: 1px solid color(grey, 100);
    box-sizing: border-box;

    input:checked + & {
        border-width: 10px;
        border-color: color(brand, primary);

        &.isDisabled {
            background-color: white;
            border-color: color(grey, 100);
        }
    }
}

.isDisabled {
    background-color: color(grey, 100);
}