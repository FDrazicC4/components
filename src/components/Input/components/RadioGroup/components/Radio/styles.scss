@import '../../../../../../styles/utilities';

.wrapper {
    display: inline-flex;
    overflow: hidden;
    align-items: center;
}

.radio {
    height: size(xx-large);
    width: size(xx-large);
    display: inline-flex;
    border-radius: 100%;
    border: 1px solid color(grey, 100);
    box-sizing: border-box;

    input:checked + & {
        border-width: 10px;
        border-color: color(brand, primary);
    }
}

.isFullWidth {
    width: 100%;
}

.isDisabled {

    input + .radio {
        background-color: color(grey, 100);
    }

    input:checked + .radio {
        background-color: white;
        border-color: color(grey, 100);
    }
}
