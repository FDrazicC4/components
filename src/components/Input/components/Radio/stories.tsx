import React, { ReactElement } from 'react';
import Radio from './index';
import Flag from '@components/Flag';
import { boolean, text } from '@storybook/addon-knobs';


export default {
    component: Radio,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Components|User Input/Radio',
};


export const Default = (): ReactElement => {
    const checked = boolean('checked', false);
    const disabled = boolean('disabled', false);
    const children = text('children', 'Radio label');
    const hasError = boolean('hasError', false);

    return (
        <Radio checked={checked} disabled={disabled} hasError={hasError}>{children}</Radio>
    );
};
