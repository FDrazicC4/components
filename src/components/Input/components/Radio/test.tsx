import React from 'react';
import Radio from './index';
import renderer from 'react-test-renderer';


describe('Radio', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<Radio />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
