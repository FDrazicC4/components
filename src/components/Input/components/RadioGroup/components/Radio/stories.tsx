import React, { ReactElement } from 'react';
import RadioGroup from '../../../RadioGroup';
import Radio from './index';
import Label from '../../../../../Label';
import { boolean, text } from '@storybook/addon-knobs';


export default {
    component: Radio,
    parameters: {
        componentSubtitle: <Label color="warning">In Development</Label>,
    },
    title: 'Components|User Input/Input/RadioGroup/Radio',
};


export const Default = (): ReactElement => {
    const checked = boolean('checked', false);
    const disabled = boolean('disabled', false);
    const fullWidth = boolean('fullWidth', false);
    const inputLabel = text('inputLabel', 'Pick an option:');
    const radioSetName = text('radioSetName', 'testing');
    const optionName1 = text('optionName1', 'testing1');
    const optionName2 = text('optionName2', 'testing2');

    return (
        <RadioGroup inputLabel={inputLabel}>
            <Radio
                name={radioSetName}
                inputLabel={optionName1}
                checked={checked}
                disabled={disabled}
                fullWidth={fullWidth}
                readOnly
            />
            <Radio
                name={radioSetName}
                inputLabel={optionName2}
                checked={checked}
                disabled={disabled}
                fullWidth={fullWidth}
                readOnly
            />
        </RadioGroup>

    );
};
