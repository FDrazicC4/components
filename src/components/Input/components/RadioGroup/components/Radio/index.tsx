import React, { ReactElement } from 'react';
import clsx from 'clsx';
import InputBase from '../../../InputBase';
import { Props } from './types';
import styles from './styles.scss';


function Radio({
    className,
    customControl,
    disabled = false,
    fullWidth = false,
    inputLabel,
    checked,
    onChange,
    ...otherProps
}: Props): ReactElement {


    const defaultCustomControl = (): ReactElement => <div className={styles.radio} />;


    const classes = clsx(
        styles.wrapper,
        { [styles.isDisabled]: Boolean(disabled) },
        { [styles.isFullWidth]: Boolean(fullWidth) },
        className,
    );


    function handleOnChange() {
        checked = !checked;
    }


    return (
        <InputBase
            className={classes}
            inputLabel={inputLabel}
            customControl={customControl || defaultCustomControl}
            disabled={disabled}
            fullWidth={fullWidth}
            type="radio"
            onChange={handleOnChange}
            {...otherProps}
        />
    );

}


export default Radio;
