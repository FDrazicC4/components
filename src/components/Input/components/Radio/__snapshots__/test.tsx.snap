// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<label
  className="brew-Typography brew-Typography--display-block brew-Input brew-Radio"
  tabIndex={1}
>
  <input
    className="brew-Input__element"
    type="radio"
  />
  <div
    className="brew-Radio__control"
  />
</label>
`;
